packages/venia-concept/src/ServiceWorker/Utilities/htmlHandler.js

```diff
@@ -35,15 +172,21 @@ export const cacheHTMLPlugin = {
          * Check if there is a response cached for the request
          * URL. If yes, compare the content of both the cached
          * response and the new response. If they are different
-         * send a the `HTML_UPDATE_AVAILABLE` message to window.
+         * send the `HTML_UPDATE_AVAILABLE` message to window.
          */
         const cachedResponseObj = await caches.match(
             new URL(request.url).pathname
         );
         if (cachedResponseObj) {
             const cachedResponse = await cachedResponseObj.text();
             const clonedResponse = await response.clone().text();
-            if (cachedResponse !== clonedResponse) {
+
+            if (
+                hasHTMLChanged(
+                    parse(clonedResponse, { style: true }),
+                    parse(cachedResponse, { style: true })
+                )
```

Comment:
We could save ourselves some time by keeping both conditionals here.

Only if `cachedResponse !== clonedResponse` do we check `hasHTMLChanged`. This way if they are the same, we don't do any of the more time-intensive checks.

Reply:
Nice catch Andy. This saves time. We can only calculate the differences if the HTMLs differ.